### ISSUE TITLE
DiCy doesn't recognize LaTeX files if they include commands before the document class

### ISSUE THREAD

**Issue by ngld** [yitzchak/ouroboros#142]
I have a LaTeX file which begins like this:
```latex
\PassOptionsToPackage{unicode=true}{hyperref} % options for packages loaded elsewhere
\PassOptionsToPackage{hyphens}{url}
\documentclass{article}
```
DiCy doesn't recognize it as a LaTeX file since the file type regex is too strict and dosen't allow any commands before `\documentclass`. Would it be possible to relax that regex?

As a workaround, I've moved the commands below `\documentclass` which fixes the problem but it would be nice if DiCy could support these files without changes.

**Comment by yitzchak** [comment]
I don't have a problem considering relaxing the regex, but is `\PassOptionsToPackage` actually required before `\documentclass`?

**Comment by ngld** [comment]
`\PassOptionsToPackage` doesn't have to appear before `\documentclass`. As I mentioned, moving it below `\documentclass` fixes the problem for me.

However, it was weird having DiCy complain that no rules were defined for my file even though it was a valid LaTeX file. After digging through the code, I found out that it didn't detect it as LaTeX due to the regex not matching.

**Comment by yitzchak** [comment]
Okay, I understand. I'll see what I can do about tweaking the regex. 

Thanks again for the feedback!

**Comment by yitzchak** [comment]
@ngld Just released v0.14.0 with this fix. Thanks for pointing out this issue.
